darshan-util/pydarshan/darshan/experimental/plots/data_access_by_filesystem.py

```diff
@@ -0,0 +1,606 @@
+
+        # scale to fit longer filesystem
+        # strings on the left side of the plots
+        # NOTE: may need more sophisticated scaling
+        # eventually
+        if len(filesystem) <= 8 and not '<STD' in filesystem:
+            fontsize = 18
+        else:
+            fontsize = 12
+
+        # anonymized STD.. streams have associated integers
+        # that are stored in the filesystem data field
+        # but that are confusing to display, so strip them
+        if filesystem.startswith('anonymized'):
+            ax_filesystem_bytes.annotate('anonymized', (-0.3, 0.5), fontsize=fontsize, xycoords='axes fraction')
+        else:
+            ax_filesystem_bytes.annotate(filesystem, (-0.3, 0.5), fontsize=fontsize, xycoords='axes fraction')
+
+        ax_filesystem_counts.barh(0, files_written, color='red', alpha=0.3)
+        ax_filesystem_counts.barh(1, files_read, color='blue', alpha=0.3)
+
+        ax_filesystem_bytes.text(0, 0.75, f' # bytes read ({bytes_read:.2E} MiB)', transform=ax_filesystem_bytes.transAxes)
+        ax_filesystem_bytes.text(0, 0.25, f' # bytes written ({bytes_written:.2E} MiB)', transform=ax_filesystem_bytes.transAxes)
+
+        if files_read == 0:
+            ax_filesystem_counts.text(0, 0.75, ' 0 files read', transform=ax_filesystem_counts.transAxes)
+        else:
+            ax_filesystem_counts.text(0, 0.75, f' # files read ({files_read:.2E})', transform=ax_filesystem_counts.transAxes)
+
+        if files_written == 0:
+            ax_filesystem_counts.text(0, 0.25, ' 0 files written', transform=ax_filesystem_counts.transAxes)
+        else:
+            ax_filesystem_counts.text(0, 0.25, f' # files written ({files_written:.2E})', transform=ax_filesystem_counts.transAxes)
+
+        ax_filesystem_bytes.barh(0, bytes_written, color='red', alpha=0.3)
+        ax_filesystem_bytes.barh(1, bytes_read, color='blue', alpha=0.3)
+
+        for axis in fig.axes:
+            # hide the right side plot frame (spine) so that
+            # the value labels don't overlap with the plot frame
+            axis.spines['right'].set_visible(False)
+            axis.set_xticklabels([])
+            axis.set_yticklabels([])
+            axis.xaxis.set_ticks_position('none')
+            axis.yaxis.set_ticks_position('none')
+            axis.minorticks_off()
+
+
+def plot_with_log_file(log_file_path: str, plot_filename: str, verbose: bool = False):
+    """
+    Plot the data access by category given a darshan log
+    file path.
+
+    Parameters
+    ----------
+
+    log_file_path: path to the darshan log file
+
+    plot_filename: name of the plot file produced
+
+    verbose: if ``True``, provide extra debug information
+
+    Returns
+    -------
+
+    fig: matplotlib figure object
+    """
+    fig = plt.figure()
+    log_file = os.path.basename(log_file_path)
+    fig.suptitle(f"Data Access by Category for log file: '{log_file}'")
+    report = darshan.DarshanReport(log_file_path, read_all=True)
```

Comment:
Based on the discussion in #487, I think we want this function to take the `DarshanReport` as an input over the `log_file_path`.  This would also mean updating the docstring with something like `report: a ``darshan.DarshanReport``.`

Also, this change would make it a little more difficult to put the `log_file` in the title, but I don't think we need that anyways since this will be in a report where `log_file` is already displayed.